src/Bicep.Core/Semantics/Binder.cs

```diff
@@ -43,6 +45,26 @@ public class Binder : IBinder
             this.Bindings = NameBindingVisitor.GetBindings(sourceFile.ProgramSyntax, NamespaceResolver, fileScope);
             this.cyclesBySymbol = CyclicCheckVisitor.FindCycles(sourceFile.ProgramSyntax, this.Bindings);
 
+            var extendsDeclarations = sourceFile.ProgramSyntax.Declarations.OfType<ExtendsDeclarationSyntax>();
+
+            foreach (var extendsDeclaration in extendsDeclarations)
+            {
+                if (sourceFileLookup.TryGetSourceFile(extendsDeclaration).TryUnwrap() is {} extendedFile &&
+                    modelLookup.GetSemanticModel(extendedFile) is SemanticModel extendedModel)
+                {
+                    var parameterAssignments = ImmutableArray<ParameterAssignmentSymbol>.Empty;
+                    foreach (var assignment in extendedModel.Root.ParameterAssignments)
+                    {
+                        if (!fileScope.Locals.Any(e => e.Name == assignment.Name))
+                        {
+                            parameterAssignments = parameterAssignments.Add(assignment);
+                        }
```

Comment:
Did we close on the behavior we wanted w.r.t overriding?

Reply:
Example where I think we need to think through the expected behavior carefully to avoid confusing people:

Base file (`base.bicepparam`)
```bicep
using none

param foo = 'foo'
param bar = foo
```

Extending file (`main.bicepparam`)
```bicep
extends 'base.bicepparam'

param foo = 'bar'
param baz = foo
```

What are the values you would expect for `foo`, `bar` and `baz`?